lib/mayaUsd/ufe/UsdUndoableCommand.h

```diff
@@ -0,0 +1,197 @@
+//
+// Copyright 2020 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#pragma once
+
+#include <mayaUsd/base/api.h>
+#include <mayaUsd/undo/UsdUndoBlock.h>
+#include <mayaUsd/undo/UsdUndoableItem.h>
+
+#include <pxr/base/vt/value.h>
+#include <pxr/usd/usdGeom/xformOp.h>
+
+#include <ufe/baseUndoableCommands.h>
+#include <ufe/path.h>
+
+PXR_NAMESPACE_USING_DIRECTIVE
+
+namespace MAYAUSD_NS_DEF {
+namespace ufe {
+
```

Comment:
This is the starting point for the refactor, the top-most new base class.

Its goals, as stated are:
- Implement the UFE undo/redo/execute protocol/virtual functions.
- Hide the peculiarity of the undo/redo protocol, for example that some commands actually call set() instead of redo(), or that undo(0 may be called before execute().
- Provide the simplest interface for sub-classes. Basically, only handleSet(0 and optionally set(0 are needed.